src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -49,6 +52,75 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
+        private static readonly object InternalSyncObject = new object();
```

Comment:
At worst wouldn't we just be synchronizing the first access that initializes any of the .Out/.In/.Error properties?  If so, doesn't seem worth an additional object, both in terms of runtime overhead and in terms of understanding what synchronization object to use for what.  But it's up to you.